### PR TITLE
Merge app code

### PR DESCRIPTION
**Pull request by Ianlaur**
- Register page (User/Company)
- Swiper function 
- Swiper page (Company/User)
- Chat page
- Profil page 
- Route page

### CODE REVIEW

**Review by Carbon-404 — COMMENTED** [review]
Interesting

**Review by Pierre2103 — APPROVED** [review]

**Review by Pierre2103 — DISMISSED** [review]
👍

File: src/adopte_un_candidat/lib/chat.dart [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+  }
+
+  @override
+  void dispose() {
+    scrollController.dispose();
+    super.dispose();
+  }
+
+  void scrollToBottom() {
+    // scrollController.addListener(() {
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/chat.dart:64]: useless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 2]
```diff
@@ -0,0 +1,979 @@
...
+          child: Container(
+            padding: const EdgeInsets.all(15),
+            decoration: BoxDecoration(
+              shape: BoxShape.rectangle,
+              borderRadius: BorderRadius.circular(25),
+              gradient: LinearGradient(
+                begin: Alignment.topCenter,
+                end: Alignment.bottomCenter,
+                colors: [
+                  // Color(int.parse("0xFF${user["colors"][0]}")),
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:71]: useless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 3]
```diff
@@ -0,0 +1,979 @@
...
+              child: reducedCardsChoice(),
+            )
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget reducedCardsChoice() {
+    // if (user["type"] == 'company') {
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:92]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 4]
```diff
@@ -0,0 +1,979 @@
...
+            child: Container(
+            padding: const EdgeInsets.all(10),
+            decoration: BoxDecoration(
+              color: Colors.white,
+              borderRadius: BorderRadius.circular(15),
+            ),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              children: [
+                // Image.network(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:108]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 5]
```diff
@@ -0,0 +1,979 @@
...
+                      mainAxisSpacing: 5,
+                      crossAxisSpacing: 5,
+                      children: [
+                        Container(
+                          alignment: Alignment.center,
+                          decoration: BoxDecoration(
+                            color: const Color(0xFFB565F4),
+                            borderRadius: BorderRadius.circular(15),
+                          ),
+                          // child: Text(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:165]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 6]
```diff
@@ -0,0 +1,979 @@
...
+                          //   ),
+                          // ),
+                        ),
+                        Container(
+                          alignment: Alignment.center,
+                          decoration: BoxDecoration(
+                            color: const Color(0xFF7DC3EA),
+                            borderRadius: BorderRadius.circular(15),
+                          ),
+                          // child: Text(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:180]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 7]
```diff
@@ -0,0 +1,979 @@
...
+                            ),
+                          ),
+                        ),
+                        Container(
+                          alignment: Alignment.center,
+                          decoration: BoxDecoration(
+                            color: const Color(0xFF7DC3EA),
+                            borderRadius: BorderRadius.circular(15),
+                          ),
+                          // child: Text(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:210]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 8]
```diff
@@ -0,0 +1,979 @@
...
+                          //   ),
+                          // ),
+                        ),
+                        Container(
+                          alignment: Alignment.center,
+                          decoration: BoxDecoration(
+                            color: const Color(0xFFB565F4),
+                            borderRadius: BorderRadius.circular(15),
+                          ),
+                          // child: Text(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:225]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 9]
```diff
@@ -0,0 +1,979 @@
...
+                    )
+                  )
+                ]
+              )
+            )
+          ),
+        ],
+      );
+    // }
+    // else {
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:254]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 10]
```diff
@@ -0,0 +1,979 @@
...
+          child: Container(
+            padding: const EdgeInsets.all(15),
+            decoration: BoxDecoration(
+              shape: BoxShape.rectangle,
+              borderRadius: BorderRadius.circular(25),
+              gradient: LinearGradient(
+                begin: Alignment.topCenter,
+                end: Alignment.bottomCenter,
+                colors: [
+                  // Color(int.parse("0xFF${user["colors"][0]}")),
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:352]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 11]
```diff
@@ -0,0 +1,979 @@
...
+              child: detailedCardsChoice(),
+            )
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget detailedCardsChoice() {
+    // if (user["type"] == 'company') {
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:373]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 12]
```diff
@@ -0,0 +1,979 @@
...
+            child: Container(
+            padding: const EdgeInsets.all(10),
+            decoration: BoxDecoration(
+              color: CupertinoColors.white,
+              borderRadius: BorderRadius.circular(15),
+            ),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceAround,
+              children: [
+                // Image.network(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:389]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 13]
```diff
@@ -0,0 +1,979 @@
...
+                //   width: 48,
+                //   height: 48,
+                // ),
+                const SizedBox(width: 10),
+                Expanded(
+                    child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  mainAxisAlignment: MainAxisAlignment.start,
+                  children: [
+                    // Text(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:400]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 14]
```diff
@@ -0,0 +1,979 @@
...
+                      mainAxisSpacing: 5,
+                      crossAxisSpacing: 5,
+                      children: [
+                        Container(
+                          alignment: Alignment.center,
+                          decoration: BoxDecoration(
+                            color: const Color(0xFFB565F4),
+                            borderRadius: BorderRadius.circular(15),
+                          ),
+                          // child: Text(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:469]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 15]
```diff
@@ -0,0 +1,979 @@
...
+                          //   ),
+                          // ),
+                        ),
+                        Container(
+                          alignment: Alignment.center,
+                          decoration: BoxDecoration(
+                            color: const Color(0xFF7DC3EA),
+                            borderRadius: BorderRadius.circular(15),
+                          ),
+                          // child: Text(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:484]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 16]
```diff
@@ -0,0 +1,979 @@
...
+                            ),
+                          ),
+                        ),
+                        Container(
+                          alignment: Alignment.center,
+                          decoration: BoxDecoration(
+                            color: const Color(0xFF7DC3EA),
+                            borderRadius: BorderRadius.circular(15),
+                          ),
+                          // child: Text(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:514]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 17]
```diff
@@ -0,0 +1,979 @@
...
+                          //   ),
+                          // ),
+                        ),
+                        Container(
+                          alignment: Alignment.center,
+                          decoration: BoxDecoration(
+                            color: const Color(0xFFB565F4),
+                            borderRadius: BorderRadius.circular(15),
+                          ),
+                          // child: Text(
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:529]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 18]
```diff
@@ -0,0 +1,979 @@
...
+                      ],
+                    )
+                  )
+                ],
+              )
+            )
+          )
+        ],
+      );
+    // } else {
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:674]: uesless comments

File: src/adopte_un_candidat/lib/favorites.dart [hunk 19]
```diff
@@ -0,0 +1,979 @@
...
+                  if (snapshot.connectionState == ConnectionState.waiting) {
+                    return const Center(
+                      child: CircularProgressIndicator(),
+                    );
+                  } else if (snapshot.hasError) {
+                    return Text('Error: ${snapshot.error}');
+                  } else {
+                    cardStack = snapshot.data;
+                    if (cardStack != null) {
+                      // savedProfilesList(context);
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/favorites.dart:962]: uesless comments

File: src/adopte_un_candidat/lib/swiper_feature.dart [hunk 20]
```diff
@@ -0,0 +1,192 @@
...
+
+  Database database = Database();
+  List<dynamic>? cardStack;
+  dynamic user;
+  int currentindex = 0;
+  String swipedirection = '';
+
+  @override
+  void initState() {
+    // Future.delayed(const Duration(seconds: 1)).then((_) {
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/swiper_feature.dart:28]: uesless comments

File: src/adopte_un_candidat/lib/swiper_feature.dart [hunk 21]
```diff
@@ -0,0 +1,192 @@
...
+    }
+    database.getStack().then((value) {
+      setState(() {
+        cardStack!.add(value);
+        currentindex = 0;
+      });
+    });
+  }
+
+  // Animates the card back and forth to teach the user that it is swipable.
```
Comment (MistzSoftware) [inline src/adopte_un_candidat/lib/swiper_feature.dart:107]: uesless comments

**Review by MistzSoftware — APPROVED** [review]

**Review by Pierre2103 — APPROVED** [review]

**Pull request merged**